File: 1164.cpp
#include <iostream>

using namespace std;

int main(){
    int n, valores, s, aux=0;
    cin >> n;
    while (true){
        if (aux == n){
            break;
        }
        cin >> valores;
        for (int i=1; i<valores; i++){
            if (valores % i == 0){
                s += i;
            }
        }
        if (s == valores){
            cout << valores << " eh perfeito" << endl;
        } else {
            cout << valores << " nao eh perfeito" << endl;
        }
        aux += 1;
        s = 0;
    }
    return 0;
}

File: 1164.py
N = int(input())
for i in range(0, N):
    numeros = int(input())
    soma = 0
    qntd = 1
    while qntd < numeros:
        if numeros % qntd == 0:
            soma += qntd
        qntd += 1
    if soma == numeros:
        print(f'{numeros} eh perfeito')
    else:
        print(f'{numeros} nao eh perfeito')